compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13712,6 +13712,20 @@ private STNode parseMatchClause() {
         STNode matchGuard = parseMatchGuard();
         STNode rightDoubleArrow = parseDoubleRightArrow();
         STNode blockStmt = parseBlockNode();
+        
+        if (isNodeListEmpty(matchPatterns)) {
+            STToken identifier = SyntaxErrors.createMissingToken(SyntaxKind.IDENTIFIER_TOKEN);
+            STNode constantPattern = STNodeFactory.createSimpleNameReferenceNode(identifier);
+            matchPatterns = STNodeFactory.createNodeList(constantPattern);
```

Comment:
Shouldn't we add this test inside `parseMatchPatternList` itself?

Reply:
We seem to do the validation outside the list in other places. 
e.g. https://github.com/ballerina-platform/ballerina-lang/blob/f2b6853b3d59b1fbb28f1d10f6be1899425c5440/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L12079-L12080

Also if we do it inside, then we won't be able to give diagnostic to the next token?